### PR TITLE
[Solved #18] Generate code coverage

### LINKED ISSUE #18: Generate code coverage
Epic:

### Description
Generate code coverage and display badges

### Tasks

- [x] Generate code coverage
- [x] Make badges to display on read me
- [x] Display code coverage on PR

### PR DESCRIPTION
**Pull request by supunTE**
Solved #18 
- [x] Generate code coverage, Implement commands to generate automatically.
- [x] Generate coverage badges
- [x] Add commands to execute on pre-push 
- [x] Show coverage logs on CI

### CODE REVIEW

**Comment by supunTE** [comment]
This PR is built on supunTE:feature/7-write-unit-tests. Therefore It has unmerged commits.

File: .github/workflows/onPR.yml [hunk 1]
```diff
@@ -20,3 +20,9 @@ jobs:
       - run: npm ci
       - run: npm run build 
       - run: npm run test
+      - run: npm run make-badges:ci
+      - run: npm run coverage:codecov
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          flags: coverage
```
Comment (pasinducw) [inline .github/workflows/onPR.yml]: Please have a newline character at the end of the last line

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+coverage:
+  status:
+    project:
+      default:
+        target: 70%
+
```
Comment (pasinducw) [inline codecov.yml]: Please have a new line character at the last line

**Review by pasinducw — CHANGES_REQUESTED** [review]

File: .github/workflows/onPR.yml (hunk 1 above)
Comment (supunTE) [reply .github/workflows/onPR.yml]: fixed

File: codecov.yml (hunk 2 above)
Comment (supunTE) [reply codecov.yml]: fixed

**Review by pasinducw — APPROVED** [review]

**Review by dalanad — APPROVED** [review]

**Pull request merged**